src/classic/clvm_tools/stages/stage_2/operators.rs

```diff
@@ -32,6 +33,7 @@ pub struct CompilerOperators {
     compile_outcomes: RefCell<HashMap<String, String>>,
     dialect: RefCell<Rc<dyn Dialect>>,
     runner: RefCell<Rc<dyn TRunProgram>>,
+    opt_memo: Rc<RefCell<HashMap<String, NodePtr>>>,
```

Comment:
the way this was built in python, there are objects that represent capabilities of the clvm interpreter and i wanted to structurally mirror that design.  they live at static scope and shouldn't be mutable.  I'm using refcell here to synchronize cache mutation.  I removed the Rc since it wasn't needed but the RefCell is needed unless I make more structural changes here.  I've been making an effort not to make too many changes in this code since it's a port the classic python compiler.

Probably i made a mistake in that, although admittedly i think trying to be very conservative in the way I made changes in here served well.  The honest truth is that I haven't been as successful as I hoped in nudging things along so maybe i should be more active in here.